lib/src/models/severity.dart

```diff
@@ -0,0 +1,51 @@
+/// Enum class for severity
+/// 
+/// Used when reporting issues
+class Severity {
+  /// Programming error
+  ///
+  /// This indicates severe issue like memory leak etc
+  /// The issue is certain
+  static const error = Severity._('error');
+
+  /// Warning
+  ///
+  /// Used for dangerous coding style that can cause severe runtime errors
+  /// For example: accessing an out of range array element
+  static const warning = Severity._('warning');
+
+  /// Performance warning
+  ///
+  /// Suboptimal code and fixing it probably leads to faster performance
```

Comment:
Correction: `Indicates a suboptimal code, fixing it will lead to a better performance`